src/eftest/report/junit.clj

```diff
@@ -0,0 +1,156 @@
+(ns eftest.report.junit
+  "A test reporter that outputs JUnit-compatible XML."
+  (:require [clojure.java.io :as io]
+            [clojure.stacktrace :as stack]
+            [clojure.test :as test]
+            [eftest.report :refer [*context*]]))
+
+(def ^:dynamic *junit-path*
+  "Path for JUnit XML output"
+  "test-results.xml")
+
+;; XML generation based on junit.clj
+
+(def ^{:private true}
+     escape-xml-map
+     (zipmap "'<>\"&" (map #(str \& % \;) '[apos lt gt quot amp])))
+
+(defn- escape-xml [text]
+  (apply str (map #(escape-xml-map % %) text)))
+
+(defn start-element
+  [tag & [attrs]]
```

Comment:
A lot of these functions could have the arguments and function name on the same like It makes things a little more concise.